lib/functions.inc.php

```diff
@@ -372,58 +536,14 @@ function change_password( $ldap, $dn, $password, $ad_mode, $ad_options, $samba_m
 
     # Get hash type if hash is set to auto
     if ( !$ad_mode && $hash == "auto" ) {
-        $search_userpassword = ldap_read( $ldap, $dn, "(objectClass=*)", array("userPassword") );
-        if ( $search_userpassword ) {
-            $userpassword = ldap_get_values($ldap, ldap_first_entry($ldap,$search_userpassword), "userPassword");
-            if ( isset($userpassword) ) {
-                if ( preg_match( '/^\{(\w+)\}/', $userpassword[0], $matches ) ) {
-                    $hash = strtoupper($matches[1]);
-                }
-            }
-        }
+        $hash = get_hash_type($ldap, $dn, $pwd_attribute);
```

Comment:
Same comment as above: I think these functions should go into https://github.com/ltb-project/ltb-ldap

Reply:
as above: see https://github.com/ltb-project/ltb-ldap/pull/7